File: VB/DxSample/Scheduler/CustomSchedulerStorage.vb
Imports Microsoft.VisualBasic
Imports System.ComponentModel
Imports System
Imports DevExpress.XtraScheduler

Namespace DxSample.Scheduler
	<ToolboxItem(True)> _
	Public Class CustomSchedulerStorage
		Inherits SchedulerStorage
		Public Sub New()
			MyBase.New()
		End Sub
		Public Sub New(ByVal components As IContainer)
			MyBase.New(components)
		End Sub

		Protected Overrides Function CreateAppointmentStorage() As AppointmentStorageBase
			Return New CustomAppointmentStorage(Me)
		End Function
	End Class
End Namespace

File: CS/DxSample/Scheduler/CustomSchedulerStorage.cs
using System.ComponentModel;
using System;
using DevExpress.XtraScheduler;

namespace DxSample.Scheduler {
    [ToolboxItem(true)]
    public class CustomSchedulerStorage :SchedulerStorage {
        public CustomSchedulerStorage() : base() { }
        public CustomSchedulerStorage(IContainer components) : base(components) { }

        protected override AppointmentStorageBase CreateAppointmentStorage() {
            return new CustomAppointmentStorage(this);
        }
    }
}
